src/transaction.js

```diff
@@ -201,11 +192,13 @@ Transaction.prototype.hashForSignature = function (inIndex, prevOutScript, hashT
 
   // in case concatenating two scripts ends up with two codeseparators,
   // or an extra one at the end, this prevents all those possible incompatibilities.
-  var hashScript = prevOutScript.without(opcodes.OP_CODESEPARATOR)
+  var hashScript = scripts.compile(scripts.decompile(prevOutScript).filter(function (x) {
```

Comment:
You lost the todo note with this replacement: 
`// FIXME: doesn't work for data chunks, maybe time to use buffertools.compare...`